### DISCUSSION TITLE
tailwind doesnt work within UI package (I have an example repo. PLS HELP)

### DISCUSSION THREAD

**Discussion by 0xsven** [vercel/turborepo#961] [Help]
I am having the following workspaces:

```
/apps
    /web
    /storybook
/packages
    /ui
```
- Both apps import the "ui package".
- The "ui package" has a dev task that starts tailwind in watch mode.
- `npm run dev` should start all three workspaces' dev modes

Problem:
tailwind doesn't work in this setup. 

Workaround:
Both following commands do work:
- `cd ./packages/ui && npm run dev`
- or ... `npm run dev -w packages/ui`

How can I make it work all with one command?

Example repo: https://github.com/0xsven/turbo-next-tailwind-storybook

**Comment by Duckinm** [comment]
I add tailwind on my root project something like; yarn add -D tailwindcss postcss autoprefixer

then put tailwind.config.js, postcss.config.js inside config

`module.exports = {
  content: ['../../packages/ui/**/*.{ts,tsx}', './src/pages/**/*.{ts,tsx}'],
  theme: {
    extend: {},
  },
  plugins: [],
}
`

`module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
`

Where ever you want to use it just create postcss.config.js and tailwind.config.js in the root of that app, but replace with this instead

`module.exports = require('config/postcss.config')
`

`module.exports = require('config/tailwind.config')
`

Assume that, you using simple nextjs as mine, I also create styles folder inside ui folder and create app.css files with @tailwind declarative, then call it like normal nextjs do in _app file like below

`import 'ui/styles/app.css'
`

don't forget to change tailwind config content as like as your folder structure! hope you can make it.

**Reply by 0xsven** [reply]
I would prefer to have the styles generation happening in `packages/ui`. Otherwise, I would have to do it in two places: `app/web` (nextjs app) and `app/storybook` (my storybook).

**Reply by FollowJack** [reply]
Did you come up with a solution?

**Reply by neznayer** [reply]
@0xsven  I'm interested as well. Did you come up with a solution?

**Reply by 0xsven** [reply]
No. I wasn't able to make this work back then.

**Reply by kibolho** [reply]
@0xsven  @neznayer  @FollowJack Seems like you need to add this configuration to your turbo.json:
```
 "dev": {
      "cache": false,
      "persistent": true,
      "dependsOn": ["^dev"],
    }
```
